### ISSUE TITLE
SNAP/LAMMPS interface (v0.1.5)

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/PyXtal_FF#31]
@David-Zagaceta @yanxon 
We need to think about this after we get the current work done.

**Comment by qzhu2017** [comment]
Let's do SNAP first, and then SO3

**Comment by qzhu2017** [comment]
@yanxon 
We need to write some documentation when this function is ready. I am thinking about the following
- [ ] how to obtain the source code and compile the MLIAP package
- [ ] An example to run the code (Geometry optimization/MD)
- [ ] An example of using the lammps calculator from python/ase

**Comment by qzhu2017** [comment]
@yanxon 

I am trying to create an example as described above (see [here](https://github.com/qzhu2017/PyXtal_FF/tree/master/examples/Si-SNAP-NN-LAMMPS)). Can you help to complete it?  I foresee several issues here.
- [ ] After I complete the run of `python snap_train.py`, the output folder has too many files.
```
qzhu@cms Si-SNAP-NN-LAMMPS (master) $ ls Si-snap/
16-16-checkpoint.pth  Stress_Train.png  Test_db.dir       TrainForce.txt     Train_db_norm.dat
Energy_Test.png       TestEnergy.txt    Test_db_norm.bak  TrainStress.txt    Train_db_norm.dir
Energy_Train.png      TestForce.txt     Test_db_norm.dat  Train_db.bak       Trainformulas.txt
Force_Test.png        TestStress.txt    Test_db_norm.dir  Train_db.dat       ase.db
Force_Train.png       Test_db.bak       Testformulas.txt  Train_db.dir       drange.npy
Stress_Test.png       Test_db.dat       TrainEnergy.txt   Train_db_norm.bak
```
Do we really need the **txt, dir, bak file**? Maybe we can clean up some unnecessary files here.

- [ ] I don't see the output of weight.txt file. Where can I get it.
- [ ] we need to provide the `in.md` file to invoke the lammps calculation.

**Comment by yanxon** [comment]
@qzhu2017 

- The dir, bak, and dat files are from shelves. The re-training won't work if we delete one of these files. If we delete one of them, we need to recalculate the descriptors.

- The txt files are useful for visualization if needed. One may need to detect outliers. Perhaps, we can dump these into a separate folder.

- You should pick `SNAP` for the descriptor in [snap_train.py](https://github.com/qzhu2017/PyXtal_FF/blob/master/examples/Si-SNAP-NN-LAMMPS/snap_train.py)

- > * [ ]  we need to provide the `in.md` file to invoke the lammps calculation.
   This works in my case. Did you update PyXtal_FF code? Perhaps, you need to delete the old versions in your python library. This occurred to me as well. It will use the last version instead of the newest one.

Btw, let me get the example script up and running. I will get this done today.

**Comment by yanxon** [comment]
@qzhu2017 

Here is my `Si-snap` folder:
```
(base) yanxon@yanxon:~/Documents/pyxtalff_test$ ls Si-snap/
16-16-checkpoint.pth  drange.npy        Force_Test.png   Stress_Test.png   Test_db.dat       Test_db_norm.dat  TestForce.txt     Train_db.bak  Train_db_norm.bak  TrainEnergy.txt    TrainStress.txt
ase.db                Energy_Test.png   Force_Train.png  Stress_Train.png  Test_db.dir       Test_db_norm.dir  Testformulas.txt  Train_db.dat  Train_db_norm.dat  TrainForce.txt
DescriptorParams.txt  Energy_Train.png  NN_weights.txt   Test_db.bak       Test_db_norm.bak  TestEnergy.txt    TestStress.txt    Train_db.dir  Train_db_norm.dir  Trainformulas.txt
```

We need two files: `DescriptorParams.txt` and `NN_weights.txt`.

**Comment by yanxon** [comment]
@qzhu2017 

I have fixed this in 3c394569f04bb6db07aa0254be1eb0bb92e75951.
Here, I also have `in.md` file running NPT simulation.

Can you please run it if it works on your side? 

Also, what do you mean by running LAMMPS with Python Wrapper? Is this similar to the FF_benchmark repo?

If it's the case, I can prepare some Jupyter notebooks for this.
